nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/command/registry/flow/ImportAllFlows.java

```diff
@@ -0,0 +1,298 @@
+
+        // Gather all flow versions and create a map for easier search by flow id
+        final Map<String, List<Integer>> versionMap = getVersionMap(client, flowMap, isInteractive);
+
+        // Create file path list
+        final List<String> files = getFilePathList(properties);
+
+        // Deserialize file content
+        final List<ImportedSnapshot> importedSnapshots = deserializeSnapshots(files);
+
+        // As we need to keep the version order the snapshot list needs to be sorted
+        importedSnapshots.sort((o1, o2) -> ComparisonChain.start()
+                .compare(o1.getSnapshot().getBucket().getName(), o2.getSnapshot().getBucket().getName())
+                .compare(o1.getSnapshot().getFlow().getName(), o2.getSnapshot().getFlow().getName())
+                .compare(o1.getSnapshot().getSnapshotMetadata().getVersion(), o2.getSnapshot().getSnapshotMetadata().getVersion())
+                .result());
+
+        for (final ImportedSnapshot snapshot : importedSnapshots) {
+
+            final String inputSource = snapshot.getInputSource();
+            final String bucketName = snapshot.getSnapshot().getBucket().getName();
+            final String bucketDescription = snapshot.getSnapshot().getBucket().getDescription();
+            final String flowName = snapshot.getSnapshot().getFlow().getName();
+            final String flowDescription = snapshot.getSnapshot().getFlow().getDescription();
+            final int flowVersion = snapshot.getSnapshot().getSnapshotMetadata().getVersion();
+            // The original bucket and flow ids must be kept otherwise NiFi won't be able to synchronize with the NiFi Registry
+            final String bucketId = snapshot.getSnapshot().getBucket().getIdentifier();
+            final String flowId = snapshot.getSnapshot().getFlow().getIdentifier();
+
+            // Create bucket if missing
+            if (bucketMap.containsKey(bucketName)) {
+                printMessage(isInteractive, bucketName + SKIPPING_BUCKET_CREATION);
+            } else {
+                createBucket(client, bucketMap, bucketName, bucketDescription, bucketId);
+            }
+
+            // Create flow if missing
+            if (flowMap.containsKey(new ImmutablePair<>(bucketId, flowName))) {
+                if (skip) {
+                    printMessage(isInteractive, flowName + SKIPPING_IMPORT);
+                    continue;
+                } else {
+                    printMessage(isInteractive, flowName + SKIPPING_FLOW_CREATION);
+                }
+            } else if (!flowCreated.containsKey(new ImmutablePair<>(bucketId, flowName))) {
+                createFlow(client, flowCreated, flowId, flowName, flowDescription, bucketId);
+            }
+
+            // Create missing flow versions
+            if (!versionMap.getOrDefault(flowId, Collections.emptyList()).contains(flowVersion)) {
+                createFlowVersion(client, inputSource, flowId);
```

Comment:
What is the expectation for handling nested versioning during a migration?

For example, `Outer PG` under version control, and inside it there is `Child PG` also under version control. In registry, the snapshot of `Outer PG` will have a process group for `Child PG` with coordinates pointing to the registry it came from:
```
"versionedFlowCoordinates" : {
        "storageLocation" : "http://localhost:18080/nifi-registry-api/buckets/876e0978-ffe5-43c5-ab2c-69dd0a8d3c52/flows/412512c1-c95f-4a2a-9df3-8a988e326df7/versions/1",
        "registryUrl" : "http://localhost:18080",
        "bucketId" : "876e0978-ffe5-43c5-ab2c-69dd0a8d3c52",
        "flowId" : "412512c1-c95f-4a2a-9df3-8a988e326df7",
        "version" : 1
      }
```

If these commands were being used to replicate/move between two different registries at different locations, then when moving `Outer PG` to the new registry, it would need to update `storageLocation` and `registryUrl` to the location of the new registry. Otherwise when a NiFi instance pulls `Outer PG` from the new registry, it will then come to 'Child PG' and try to pull it from the old registry, or fail because no registry client exists for the URL of the old registry.